src/main/java/lotto/domain/EarningRate.java

```diff
@@ -1,46 +1,27 @@
 package lotto.domain;
 
+import lotto.utils.ValidationUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
 
 public class EarningRate {
-    private static final int RATE_NUMBER = 100;
-    private static final String ILLEGAL_PURCHASE_AMOUNT = "계산 오류입니다.Nan 또는 Infinite 발생";
-
-    private double earningRate;
-    private double totalWinningMoney = 0.0;
-
-    private int calculateEarningRate(PurchaseAmount purchaseAmount) {
-        return (int) (totalWinningMoney / purchaseAmount.getPurchaseAmount() * RATE_NUMBER);
-    }
-
-    private void generateWinningRank(List<WinningRank> winningRanks) {
-        int count;
-
-        for (WinningRank winningRank : WinningRank.values()) {
-            count = countRankPeople(winningRanks, winningRank);
 
-            sumWinningMoney(winningRank.getWinningMoney() * count);
-        }
-    }
+    private static final double RATE = 100.0;
 
-    private void sumWinningMoney(double totalWinningMoney) {
-        this.totalWinningMoney += totalWinningMoney;
-    }
+    public static double calculateEarningRate(Map<Rank, Long> eachRankCount, Money money) {
+        double totalMoney = 0.0;
+        List<Rank> rank = new ArrayList<>(Arrays.asList(Rank.values()));
+        rank.remove(Rank.NO_RANK);
 
-    public EarningRate(List<WinningRank> winningRanks, PurchaseAmount purchaseAmount) {
-        generateWinningRank(winningRanks);
-        if (Double.isInfinite(calculateEarningRate(purchaseAmount))
-                || Double.isNaN(calculateEarningRate(purchaseAmount))) {
-            throw new IllegalArgumentException(ILLEGAL_PURCHASE_AMOUNT);
+        for (Rank r : rank) {
```

Comment:
`Rank r` 축약어는 지양해 주세요!

Reply:
`List<Rank> rank` 를 ranks로 선언 하면 될 것 같아요 :) 
```suggestion
        for (Rank rank : ranks) {
```